synapse/cli/streaming.py

```diff
@@ -56,6 +60,18 @@ def read(args):
         )
         assert stream_out is not None, "No StreamOut node found in config"
 
+        if args.num_ch:
```

Comment:
I think if there is no config argument then the device shouldnt be reconfigured at all. I think of the num_ch argument as a QoL for configuring the device rather than a stand-alone option